pkg/services/sqlstore/permissions/dashboard.go

```diff
@@ -77,37 +78,62 @@ func (d DashboardPermissionFilter) Where() (string, []interface{}) {
 }
 
 type AccessControlDashboardPermissionFilter struct {
-	User            *models.SignedInUser
-	PermissionLevel models.PermissionType
+	User             *models.SignedInUser
+	dashboardActions []string
+	folderActions    []string
 }
 
-func (f AccessControlDashboardPermissionFilter) Where() (string, []interface{}) {
-	folderActions := []string{dashboards.ActionFoldersRead}
-	dashboardActions := []string{accesscontrol.ActionDashboardsRead}
-	if f.PermissionLevel == models.PERMISSION_EDIT {
-		folderActions = append(folderActions, accesscontrol.ActionDashboardsCreate)
-		dashboardActions = append(dashboardActions, accesscontrol.ActionDashboardsWrite)
+// NewAccessControlDashboardPermissionFilter creates a new AccessControlDashboardPermissionFilter that is configured with specific actions calculated based on the models.PermissionType and query type
+func NewAccessControlDashboardPermissionFilter(user *models.SignedInUser, permissionLevel models.PermissionType, queryType string) AccessControlDashboardPermissionFilter {
+	needEdit := permissionLevel > models.PERMISSION_VIEW
+	var folderActions []string
+	var dashboardActions []string
+	if queryType == searchstore.TypeAlertFolder {
+		folderActions = append(folderActions, accesscontrol.ActionAlertingRuleRead)
+		if needEdit {
+			folderActions = append(folderActions, accesscontrol.ActionAlertingRuleUpdate)
+		}
```

Comment:
As discussed in the meeting yesterday, I think we still need to add `dashboards.ActionFoldersRead` for this query. @kalleep to confirm :)